### DISCUSSION TITLE
Is DiceFocalLoss a good option for multi-class

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Project-MONAI/MONAI#2714] [Q&A]
Greetings, I want to know can DiceFocalLoss be used for multiclass because it used binary cross-entropy and also used sigmoid function? 
Has anyone experienced comparing to DiceCELoss, is DiceFocalLoss a good option for multi-class?

**Comment by Nic-Ma** [comment]
Hi @yiheng-wang-nv ,

Could you please help share some experience here? I think you may use these losses during your training before.

Thanks in advance.

**Reply by talhaanwarch** [reply]
@ahatamiz Please can you share your experienced, if you have tried DiceFocalLoss ?

**Reply by ahatamiz** [reply]
Hi @talhaanwarch,

The DiceFocalLoss can also use softmax as the activation function ( more details [here](https://docs.monai.io/en/latest/_modules/monai/losses/dice.html#DiceFocalLoss) ). I previously used it for multi-label brain tumor segmentation and achieved good results. Essentially, the focal loss should help for pixel-wise voxel/label classification. 

For more details, I recommend checking the following paper which applies this loss for whole volume head and neck segmentation:

[AnatomyNet: Deep Learning for Fast and Fully Automated Whole-volume Segmentation of Head and Neck Anatomy  ](https://arxiv.org/pdf/1808.05238.pdf)

Thanks

**Reply by talhaanwarch** [reply]
thanks @ahatamiz I have few questions. if we use softmax function in dice loss, should not we use categorical cross-entropy in focal loss. but the [implementation ](https://docs.monai.io/en/latest/_modules/monai/losses/focal_loss.html#FocalLoss) used binary cross-entropy implementation. 
second, you have used it for multi-label which means the labels are overlapped? and it is not multiclass.

**Reply by ahatamiz** [reply]
@talhaanwarch , Sure. Just to clarify my earlier response, the Softmax activation function is be used with DiceFocalLoss but only applied for Dice and not the Focal loss. 

Regarding second question, the brain tumor segmentation task (e.g. BRATS) is still considered a multi-class segmentation problem but with overlapping structures ( Sigmoid activation applied per channel). Considering this, DiceFocal is a good choice for these type of problems. Theoretically, it should also be applicable to other segmentation types (e.g. multi-organ segmentation), but I believe DiceCE should work better.

**Reply by talhaanwarch** [reply]
thanks a lot, its clear now
